Odata-docs/client/v6/deal-with-server-driven-paging.md

```diff
@@ -19,15 +19,127 @@ Example:
+        Console.WriteLine("Person Page {0}:", ++pageCount);
+
+        // If nextLink is not null, then there is a new page to load.
+        if (nextLink != null)
+        {
+            // Load the new page from the next link URI.
+            response = await context.ExecuteAsync<Person>(nextLink)
+                as QueryOperationResponse<Person>;
+        }
+
+        // Enumerate the Person(s) in the response.
+        foreach (Person p in response)
+        {
+            Console.WriteLine("\tPerson Name: {0}", p.FirstName);
+            Console.WriteLine("\tTrips Page {0}:", ++innerPageCount);
+            // Get the next link for the collection of related Trips.
+            DataServiceQueryContinuation nextTripsLink =
+                response.GetContinuation(p.Trips);
+
+            foreach (Trip t in p.Trips)
+            {
+                // Print out the trips in the first page.
+                Console.WriteLine("\t\tTripID: {0} - Name: ${1}",
+                    t.TripId, t.Name);
+            }
+
+            while (nextTripsLink != null)
+            {
+                Console.WriteLine("\t\t******Load Trips******");
+
+                // Load the next page of Trips.
+                var tripsResponse = await context.LoadPropertyAsync(p, "Trips", nextTripsLink);
+                nextTripsLink = tripsResponse.GetContinuation();
+                foreach (Trip t in tripsResponse)
+                {
+                    // Print out the trips.
+                    Console.WriteLine("\t\tTripID: {0} - Name: ${1}",
+                        t.TripId, t.Name);
+                }
+            }
+        }
+    }
+
+    // Get the next link, and continue while there is a next link.
+    while ((nextLink = response.GetContinuation()) != null);
+}
+catch (DataServiceQueryException ex)
+{
+    throw new ApplicationException(
+        "An error occurred during query execution.", ex);
+}
+```
+
+Below is the sample output (Assuming the Trippin service had a pagesize of 2):
+```html
+Person Page 1:
+        Person Name: Russell
+        Trips Page 1:
+                TripID: 0 - Name: $Trip in US
+                TripID: 1003 - Name: $Trip in Beijing
+
+        Trips Page 2:
+                TripID: 1007 - Name: $Honeymoon
+
+        Person Name: Scott
+        Trips Page 2:
+                TripID: 0 - Name: $Trip in US
+                TripID: 2004 - Name: $Trip in Beijing
+        .............................................
+        .............................................
+        .............................................
+
+Person Page 2:
+        Person Name: Marshall
+        Trips Page 3:
+               .............................................
+               ............................................
+
+        Person Name: Ryan
+        Trips Page 4:
+               .............................................
+               .............................................
+
+        Trips Page 5:
```

Comment:
I assume each person's trips page should start at Page 1. Is this "global" page counting intentional.

Reply:
#Resolved